x-pack/plugins/infra/public/components/waffle/node_context_menu.tsx

```diff
@@ -0,0 +1,198 @@
+import { useKibana } from '../../../../../../src/plugins/kibana_react/public';
+import { InventoryItemType } from '../../../common/inventory_models/types';
+import {
+  Section,
+  SectionLinkProps,
+  ActionMenu,
+  SectionTitle,
+  SectionSubtitle,
+  SectionLinks,
+  SectionLink,
+} from '../../../..//observability/public';
+
+interface Props {
+  options: InfraWaffleMapOptions;
+  currentTime: number;
+  node: InfraWaffleMapNode;
+  nodeType: InventoryItemType;
+  isPopoverOpen: boolean;
+  closePopover: () => void;
+  popoverPosition: EuiPopoverProps['anchorPosition'];
+}
+
+export const NodeContextMenu: React.FC<Props> = ({
+  options,
+  currentTime,
+  children,
+  node,
+  isPopoverOpen,
+  closePopover,
+  nodeType,
+  popoverPosition,
+}) => {
+  const uiCapabilities = useKibana().services.application?.capabilities;
+  const getUrlForApp = useKibana().services.application?.getUrlForApp;
+  const prependBasePath = useKibana().services.http?.basePath.prepend;
+  const prefixPathWithBasePath = useCallback(
+    (app: string, path?: string) => {
+      if (!getUrlForApp || !prependBasePath) {
+        return;
+      }
+      return prependBasePath(getUrlForApp(app, { path }));
+    },
+    [getUrlForApp, prependBasePath]
+  );
+  const inventoryModel = findInventoryModel(nodeType);
+  const nodeDetailFrom = currentTime - inventoryModel.metrics.defaultTimeRangeInSeconds * 1000;
+  // Due to the changing nature of the fields between APM and this UI,
+  // We need to have some exceptions until 7.0 & ECS is finalized. Reference
+  // #26620 for the details for these fields.
+  // TODO: This is tech debt, remove it after 7.0 & ECS migration.
+  const apmField = nodeType === 'host' ? 'host.hostname' : inventoryModel.fields.id;
+
+  const showDetail = inventoryModel.crosslinkSupport.details;
+  const showLogsLink =
+    inventoryModel.crosslinkSupport.logs && node.id && uiCapabilities?.logs?.show;
+  const showAPMTraceLink =
+    inventoryModel.crosslinkSupport.apm && uiCapabilities?.apm && uiCapabilities?.apm.show;
+  const showUptimeLink =
+    inventoryModel.crosslinkSupport.uptime && (['pod', 'container'].includes(nodeType) || node.ip);
+
+  const inventoryId = useMemo(() => {
+    if (nodeType === 'host') {
+      if (node.ip) {
+        return { label: <EuiCode>host.ip</EuiCode>, value: node.ip };
+      }
+    } else {
+      if (options.fields) {
+        const { id } = findInventoryFields(nodeType, options.fields);
+        return {
+          label: <EuiCode>{id}</EuiCode>,
+          value: node.id,
+        };
+      }
+    }
+    return { label: '', value: '' };
+  }, [nodeType, node.ip, node.id, options.fields]);
+
+  const nodeLogsMenuItem: SectionLinkProps = {
+    label: i18n.translate('xpack.infra.nodeContextMenu.viewLogsName', {
+      defaultMessage: '{inventoryName} logs',
+      values: { inventoryName: inventoryModel.singularDisplayName },
+    }),
+    href: getNodeLogsUrl({
```

Comment:
For me this puts a duplicate space prefix into the URL:

![grafik](https://user-images.githubusercontent.com/973741/74457041-0551ff00-4e88-11ea-9829-60c4451fa6c0.png)

That also applies to the other links in the menu.

Reply:
👍 Fixed with `prefixPathWithBasepath` changes.